puppet/lib/puppet/util/puppetdb/char_encoding.rb

```diff
@@ -32,66 +32,48 @@ module CharEncoding
   Utf8ReplacementChar = [ 0xEF, 0xBF, 0xBD ].pack("c*")
 
   DEFAULT_INVALID_CHAR = "\ufffd"
-
-  # @api private
-  def self.all_indexes_of_char(str, char)
-    (0..str.length).find_all{ |i| str[i] == char}
-  end
+  NOT_INVALID_REGEX = Regexp.new( "[^" + DEFAULT_INVALID_CHAR + "]" )
 
   # @api private
   #
-  # Takes an array and returns a sub-array without the last element
+  # Finds the beginning and ending index of the first block of invalid
+  # characters.
   #
-  # @return [Object]
-  def self.drop_last(array)
-    array[0..-2]
-  end
-
-  # @api private
-  #
-  # Takes an array of increasing integers and collapses the sequential
-  # integers into ranges
-  #
-  # @param index_array an array of sorted integers
-  # @return [Range]
-  def self.collapse_ranges(index_array)
```

Comment:
This code can be removed because we're no longer outputting information on all chunks of bad data, only the first, so there will only ever be 1 range.